### ISSUE TITLE
Expired Product details. Please fetch product details again and use it to retry the call

### ISSUE THREAD

**Issue by lihaoliangGz** [android/play-billing-samples#616]
On PurchasesUpdatedListener.onPurchasesUpdated() calback, BillingResult.getResponseCode() is "5",  BillingResult.getDebugMessage() is
"Expired Product details. Please fetch product details again and use it to retry the call.".

The SkuDetails that pull up google Pay are cached in memory.
The cached SkuDetails are expired , right? How long is the expiration period?

**Comment by chootao** [comment]
> Did you solve it, I had the same problem

**Comment by cotfas** [comment]
Any news on this?

**Comment by LifeIsJourney** [comment]
Got the same issue.
Fixed by clearing the cache of the Play Store on my device.

**Comment by cotfas** [comment]
Yes, it was the same for me, i got rid of that error by clearing cache of google play store on the device. But most probably if this happens in production for an user, will be hard to detect and tell the user/customer to do the clearing cache.

**Comment by weiwenjuan** [comment]
Yes, it was the same for me, i got rid of that error by clearing cache of google play store on the device. But most probably if this happens in production for an user, will be hard to detect and tell the user/customer to do the clearing cache.
